src/snovault/resources.py

```diff
@@ -168,7 +168,12 @@ def get(self, name, default=None):
                 return default
             return resource
         if self.unique_key is not None:
-            resource = self.connection.get_by_unique_key(self.unique_key, name)
+            # Give the storage a hint of which index to search.
+            # If this is a collection of an abstract type,
```

Comment:
Now, THAT I did not consider.   We would have to register abstract types more firmly if we were to map an abstract class to all it's subclasses.   I'll think about this as a possible enhancement.

Reply:
Given that it's a performance optimization and looking up specific items within an abstract collection is probably uncommon, I doubt it's very high priority.